### ISSUE TITLE
Support for MPS backend

### ISSUE THREAD

**Issue by Tianyuan-Zhang** [khalil-research/PyEPO#43]
The current version doesn't support for the MPS backend which is available in PyTorch.
Is is possible to add support for MPS backend?
I tried to manually move the tensors and predictive models to MPS device, but got errors when calculating the loss functions.

**Comment by LucasBoTang** [comment]
Hi Tianyuan,

PyEPO hasn't been tested on the MPS backend. Generally, the device should not affect the calculation of loss functions, but it would help if you could provide the specific error messages you're encountering so I can examine the issue more closely.

If you're comfortable sharing your code, I can also try to test it. However, I don't currently have access to a macOS system currently.

**Comment by Tianyuan-Zhang** [comment]
[mps_bug.zip](https://github.com/user-attachments/files/17432404/mps_bug.zip)
I've attached two Jupyter notebooks that show the error.
The first one, I didn't move the tensors to the MPS device, which caused an error when calculating the regret.
The second one, I moved the tensors to the MPS device, but still got an error when calculating the regret.

**Comment by LucasBoTang** [comment]
Thank you for sharing the details and attaching the Jupyter notebooks related to the MPS device issue. I apologize for the delay in following up on this.

Could you let me know if this issue has been resolved on your end, or if it's still persisting? If not, I’d be happy to investigate it further. Please confirm if you’re still encountering the error, and I’ll prioritize looking into it.
